test/github_pullrequest_test.go

```diff
@@ -135,6 +137,57 @@ func TestGithubPullRequestSecondBadYaml(t *testing.T) {
 	golden.Assert(t, res.CheckRuns[0].GetOutput().GetText(), strings.ReplaceAll(fmt.Sprintf("%s.golden", t.Name()), "/", "-"))
 }
 
+// TestGithubPullRequestInvalidSpecValues tests invalid field values of a PipelinRun and
+// ensures that these validation errors are reported on UI.
+func TestGithubPullRequestInvalidSpecValues(t *testing.T) {
+	ctx := context.Background()
+	g := &tgithub.PRTest{
+		Label:            "Github Rerequest",
+		YamlFiles:        []string{"testdata/failures/invalid-timeouts-values-pipelinerun.yaml"},
+		SecondController: true,
+		NoStatusCheck:    true,
+	}
+	g.RunPullRequest(ctx, t)
+	defer g.TearDown(ctx, t)
+
+	opt := github.ListOptions{}
+	res := &github.ListCheckRunsResults{}
+	resp := &github.Response{}
+	var err error
+	counter := 0
+	for {
+		res, resp, err = g.Provider.Client.Checks.ListCheckRunsForRef(ctx, g.Options.Organization, g.Options.Repo, g.SHA, &github.ListCheckRunsOptions{
+			AppID:       g.Provider.ApplicationID,
+			Status:      github.String("completed"),
+			ListOptions: opt,
+		})
+		assert.NilError(t, err)
+		assert.Equal(t, resp.StatusCode, 200)
+		if len(res.CheckRuns) > 0 {
+			break
+		}
+		g.Cnx.Clients.Log.Infof("Waiting for the check run to be created")
+		if counter > 10 {
+			t.Errorf("Check run not created after 10 tries")
+			break
+		}
+		time.Sleep(5 * time.Second)
+	}
+
+	assert.Equal(t, len(res.CheckRuns), 1)
+	assert.Equal(t, res.CheckRuns[0].GetOutput().GetTitle(), "pipelinerun start failure")
+	reg := regexp.MustCompile("Pipelines as Code.* has <b>failed</b>.")
+	assert.Assert(t, cmp.Regexp(reg, res.CheckRuns[0].GetOutput().GetSummary()))
+	// Regular expression to match description of status of PipelineRun, as invalid `.spec.timeouts`
+	// values are being matched here, no specific duration value is used instead general words ([\w])
+	// and digits ([\d]) patterns are used.
+	pattern := `invalid value: ([\d\w]+) should be <= pipeline duration: ([\w\.\/\-]+)(?:, ([\w\.\/\-]+))?` +
+		`|invalid value: ([\d\w]+) \+ ([\d\w]+) should be <= pipeline duration: ([\w\.\/\-]+), ([\w\.\/\-]+)` +
+		`|invalid value: ([\d\w]+) should be <= pipeline duration: ([\w\.\/\-]+)`
```

Comment:
Can you put the pattern in common place so that can be referred from both GitHub and Gitea

Reply:
I think we can put under options.go